set.mm

```diff
@@ -555147,13 +555146,340 @@ relation with an intersection with the same Cartesian product (cf.
+    LUIUAUCQZAGUEUHUJABCMNUAUCAOPBUFMS $.
+
+  $( Two ways to express restriction of range Cartesian product, cf. ~ xrnres ,
+     ~ xrnres2 .  (Contributed by Peter Mazsa, 28-Mar-2020.) $)
+  xrnres3 $p |- ( ( R |X. S ) |` A ) = ( ( R |` A ) |X. ( S |` A ) ) $=
+    ( c1st cvv cxp cres ccnv ccom c2nd cin cxrn ineq12i df-xrn reseq1i resindir
+    resco eqtri 3eqtr4i ) DEEFZGHZBIZAGZJTGHZCIZAGZKZUABAGZIZUDCAGZIZKBCLZAGZUH
+    UJLUCUIUFUKUABAQUDCAQMUMUBUEKZAGUGULUNABCNOUBUEAPRUHUJNS $.
+
+  $( Two ways to express restriction of range Cartesian product.  (Contributed
+     by Peter Mazsa, 29-Dec-2020.) $)
+  xrnres4 $p |- ( ( R |X. S ) |` A ) =
+      ( ( R |X. S ) i^i ( A X. ( ran ( R |` A ) X. ran ( S |` A ) ) ) ) $=
+    ( cxrn cres crn cxp cin xrnres3 dfres4 xrneq12i inxpxrn 3eqtri ) BCDZAEBAEZ
+    CAEZDBAOFZGHZCAPFZGHZDNAQSGGHABCIORPTABJACJKAQSBCLM $.
+
+  $( Sethood condition for the restricted range cross of classes.  (Contributed
+     by Peter Mazsa, 16-Dec-2020.)  (Revised by Peter Mazsa, 7-Sep-2021.) $)
+  xrnresex $p |- ( ( A e. V /\ R e. W /\ ( S |` A ) e. X ) ->
+                   ( R |X. ( S |` A ) ) e. _V ) $=
+    ( wcel cres w3a cxrn cvv crn cxp cin simp1 resexg rnexg syl dfres4 eleq1i
+    3ad2ant2 3ad2ant3 inxpxrn xrninxpex syl5eqel xrneq12i sylibr xrnres3 eqtr3i
+    3jca xrnres2 sylib ) ADGZBEGZCAHZFGZIZBAHZUOJZKGZBUOJZKGUQBAURLZMNZCAUOLZMN
+    ZJZKGZUTUQUMVBKGZVDKGZIZVGUQUMVHVIUMUNUPOUNUMVHUPUNURKGVHBAEPURKQRUAUPUMVIU
+    NUOFQUBUJVJVFBCJZAVBVDMMNKAVBVDBCUCAVBVDBCDKKUDUERUSVFKURVCUOVEABSACSUFTUGU
+    SVAKVKAHUSVAABCUHABCUKUITUL $.
+
+  $( Sethood condition for the range cross restricted identity.  (Contributed
+     by Peter Mazsa, 31-Dec-2021.) $)
+  xrnidresex $p |- ( ( A e. V /\ R e. W ) ->
+                     ( R |X. ( _I |` A ) ) e. _V ) $=
+    ( wcel wa cid cres cvv w3a cxrn resiexg adantr ancli df-3an sylibr xrnresex
+    syl ) ACEZBDEZFZSTGAHZIEZJZBUBKIEUAUAUCFUDUAUCSUCTACLMNSTUCOPABGCDIQR $.
+
+  $( Sethood condition for the range cross restricted converse epsilon.
+     (Contributed by Peter Mazsa, 16-Dec-2020.)  (Revised by Peter Mazsa,
+     23-Sep-2021.) $)
+  xrncnvepresex $p |- ( ( A e. V /\ R e. W ) ->
+                        ( R |X. ( `' _E |` A ) ) e. _V ) $=
+    ( wcel cep ccnv cres cvv w3a cnvepresex adantr ancli df-3an sylibr xrnresex
+    wa cxrn syl ) ACEZBDEZQZTUAFGZAHZIEZJZBUDRIEUBUBUEQUFUBUETUEUAACKLMTUAUENOA
+    BUCCDIPS $.
+
+  $( Binary relation on an intersection is a special case of binary relation on
+     range Cartesian product.  (Contributed by Peter Mazsa, 21-Aug-2021.) $)
+  brin2 $p |- ( ( A e. V /\ B e. W ) ->
+                ( A ( R i^i S ) B <-> A ( R |X. S ) <. B , B >. ) ) $=
+    ( wcel wa cop cxrn wbr cin wb brxrn 3anidm23 brin syl6rbbr ) AEGZBFGZHABBIC
+    DJKZABCKABDKHZABCDLKRSTUAMABBCDEFFNOABCDPQ $.
+
+  $( Binary relation on an intersection is a special case of binary relation on
+     range Cartesian product.  (Contributed by Peter Mazsa, 21-Aug-2021.) $)
```

Comment:
@benoit thank you for your comments, I´ll resolve them asap, tomorrow at night.

Reply:
I see `~df-op` and many depending theorems have a tag "_(Avoid depending on this detail.)_". We could probably propagate that tag to new theorems: `~opidORIG`, `~brin3`.

Reply:
Looking at this I stumbled upon `~opideq`, which also relies on `~opidORIG`.
There must be a much shorter proof for that theorem, using `~opthg`, and does not need to rely on "that detail".

Reply:
I agree with @tirix on both remarks: opidORIG and brin3, if not removed, should have that tag. But the best way not to propagate these construction-dependent results is not to have them. The specifics of the encoding of the ordered pair should be used only for the most basic theorems immediately following its definition, but not later. The defining result for ordered pairs is opthg, which should suffice for almost everything, together with the results about 1st and 2nd (and indeed opideq should be prove from opthg).
Note: @avekens's opidg should also have that tag, and probably be move to Main and used to prove opid. Can you do it, @avekens ?

Reply:
Since opidg is in my mathbox, I will add the tag and move it to main with my next PR. In the same PR, I will delete and replace opidORIG of Peter's Mathbox.

Reply:
Actually, the proof of ~opideq becomes much shorter using ~opthg, and the antecedent B e. W can be removed. Since this theorem is not used (yet) I will change it with my next PR, too. @mazsa  OK?

Reply:
Alexander @avekens , I'm back. I'va done it this way, what is your opinion (note ( A e. V \/ B e. W ) !):
  $( Equality conditions for ordered pairs ` <. A , A >. ` and
     ` <. B , B >. ` .  (Contributed by Peter Mazsa, 22-Jul-2019.)  (Revised by
     Thierry Arnoux, 16-Feb-2022.) $)
  opideq $p |- ( ( A e. V \/ B e. W ) ->
                 ( <. A , A >. = <. B , B >. <-> A = B ) ) $=
    ( wcel cop wceq wb wa opthg anidms anidm syl6bb eqcom 3bitr3g jaoi ) ACEZAA
    FZBBFZGZABGZHBDEZQTUAUAIZUAQTUCHAABBCCJKUALMUBSRGZBAGZTUAUBUDUEUEIZUEUBUDUF
    HBBAADDJKUELMSRNBANOP $.

Reply:
I have a shorter proof: 
( ( A e. V /\ B e. W ) -> ( <. A , A >. = <. B , B >. <-> A = B ) ) $=
    ( wcel wa cop wceq wb simpl jca opthg syl anidm a1i bitrd ) ACEZBDEZFZAAGBB
    GHZABHZUAFZUASQQFTUBISQQQRJZUCKAABBCCLMUBUAISUANOP $.

And as I wrote before, B e. W is not required, so you can prove:

opideq $p |- ( A e. V -> ( <. A , A >. = <. B , B >. <-> A = B ) ) $=
    ( wcel cop wceq wa wb opthg anidms anidm syl6bb ) ACDZAAEBBEFZABFZOGZOMNPHA
    ABBCCIJOKL $.

Reply:
Hi @mazsa,
You could do even shorter like this:
```metamath
  $( Equality conditions for ordered pairs ` <. A , A >. ` and
     ` <. B , B >. ` .  (Contributed by Peter Mazsa, 22-Jul-2019.)  (Revised by
     Thierry Arnoux, 16-Feb-2022.) $)
  opideqALT $p |- ( A e. V -> ( <. A , A >. = <. B , B >. <-> A = B ) ) $=
    ( wcel wa cop wceq wb pm4.24 opthg anidm syl6bb sylbi ) ACDZNNEZAAFBBFGZABG
    ZHNIOPQQEQAABBCCJQKLM $.
```
I guess this is what @avekens  is referring to.

Reply:
Hi @tirix yes of course, and as a revisor you should decide which dou yo prefer. Should I change it? (or if somebody merges this PR in between, @avekens will do). However, there is a deeper question / design choice here. We have quite some similar theorems around where we publish first a theorem $p |- ( A e. V -> blabla ) then, later on, somebody needs the same blabla with different sufficient condition and we publish the theorem2 $p |- ( B e. W -> blabla ) or reltheorem $p |- ( Rel R -> blabla ). This is 2 or more theorems, and you can consider 1 theorem with $p |- ( ( A e. V \/ B e. W ) -> blabla ) shorter than these 2. Perhaps this question deserves a github issue later on.

Reply:
> Hi @tirix yes of course, and as a revisor you should decide which do you prefer. Should I change it? (or if somebody merges this PR in between, @avekens will do). 

I have prepared it already for my next PR (coming soon, maybe today or tomorrow).

>However, there is a deeper question / design choice here. We have quite some similar theorems around where we publish first a theorem $p |- ( A e. V -> blabla ) then, later on, somebody needs the same blabla with different sufficient condition and we publish the theorem2 $p |- ( B e. W -> blabla ) or reltheorem $p |- ( Rel R -> blabla ). This is 2 or more theorems, and you can consider 1 theorem with `$p |- ( ( A e. V \/ B e. W ) -> blabla )` shorter than these 2. Perhaps this question deserves a github issue later on.

These are different questions, depending on an `/\`or `\/` antecedent: with `$p |- ( ( A e. V \/ B e. W ) -> blabla )` you are right, but with `$p |- ( ( A e. V /\ B e. W ) -> blabla )`, as in the current case, either we need always both conditions, or only one of them is necessary and sufficient (say `A e. V ` ) then `$p |- ( A e. V -> blabla )` is sufficient, and `$p |- ( B e. W -> blabla )` would simply not hold.

Reply:
> with `$p |- ( ( A e. V /\ B e. W ) -> blabla )`, as in the current case, either we need always both conditions, or only one of them is necessary and sufficient (say `A e. V ` ) then `$p |- ( A e. V -> blabla )` is sufficient, and `$p |- ( B e. W -> blabla )` would simply not hold.

@avekens the original was `$p |- ( ( A e. V /\ B e. W ) -> blabla )` in fact, then came @tirix with his revision and I committed the `$p |- ( ( A e. V \/ B e. W ) -> blabla )` version 
https://github.com/metamath/set.mm/commit/b9cd7f5d50ff4d6ff206272ad6565b784f16dab1
this is it:
```metamath
50::opthg          |- (  ( A e. V /\ A e. V )
                      -> ( <. A , A >. = <. B , B >. <-> ( A = B /\ A = B ) ) )
51:50:anidms       |- (  A e. V
                      -> ( <. A , A >. = <. B , B >. <-> ( A = B /\ A = B ) ) )
52::anidm          |- ( ( A = B /\ A = B ) <-> A = B )
53:51,52:syl6bb    |- (  A e. V
                      -> ( <. A , A >. = <. B , B >. <-> A = B ) )
54::opthg          |- (  ( B e. W /\ B e. W )
                      -> ( <. B , B >. = <. A , A >. <-> ( B = A /\ B = A ) ) )
55:54:anidms       |- (  B e. W
                      -> ( <. B , B >. = <. A , A >. <-> ( B = A /\ B = A ) ) )
56::anidm          |- ( ( B = A /\ B = A ) <-> B = A )
57:55,56:syl6bb    |- (  B e. W
                      -> ( <. B , B >. = <. A , A >. <-> B = A ) )
58::eqcom          |- (   <. B , B >. = <. A , A >.
                      <-> <. A , A >. = <. B , B >. )
59::eqcom          |- ( B = A <-> A = B )
60:57,58,59:3bitr3g 
                   |- (  B e. W
                      -> ( <. A , A >. = <. B , B >. <-> A = B ) )
qed:53,60:jaoi     |- (  ( A e. V \/ B e. W )
                      -> ( <. A , A >. = <. B , B >. <-> A = B ) )
```

Reply:
Ok! I've added my approval to this PR.
@avekens, please move on with what you have in stock, thank you so much!

As for ` ( A e. V -> ` only or also ` ( B e. V -> `, I'd say we shall see the use case, i.e. rather go first with only the first one, and add the second if needed.

Reply:
@tirix ok I committed the ` ( A e. V -> ` version as you suggested. @avekens you should change it or any part of it as you think but I need it later so please keep it in my box or move to the main + as I remember you said that you will delete opidORIG in my box, didn´t you? Please do it and modify brin3 accordingly as well. Thank you.

Reply:
@mazsa I created a PR with changes in your mathbox, as discussed. I fear that there will be merge conflicts with my or your PR, depending in which order they are merged. I think, however, they can be solved without much effort.